.rubocop.yml

```diff
@@ -18,6 +18,12 @@ AllCops:
     - 'truffleruby/target/classes/**/*'
     - 'mxbuild/**/*'
 
+Lint/AssignmentInCondition:
+  Description: Don't use assignment in conditions.
+  StyleGuide: "#safe-assignment-in-condition"
+  Enabled: true
+  AllowSafeAssignment: true
+
```

Comment:
The improve consistency I had in mind comes from a fact that if `(=)` is enforced for assignments in conditions then when reading a condition's source code which is without any brackets (easy to spot) a reader can be immediately sure that there is no assignment in the condition. This is not possible without the rule, a reader has to scan each condition for `=/==` which is harder than spotting just `()`. Of course `()` don't imply assignment, but a condition without `()` is still frequent enough to make the rule valuable.

```ruby
# For comparison
while (item = queue.pop)
while item = queue.pop
while item == queue.pop
```

The cop does not prevent using `while item = queue.pop` which is indeed a nice pattern, it only enforces adding brackets `while (item = queue.pop)`.

I also think that it's better to make assignments outside of the conditions, but in a cases when it might make sense to have it in the condition (loop, elsif) I believe it's valuable for the assignment to be more distinguishable. 

We can have this enabled to improve the visibility of assignments, and also move the assignments out of the conditions if we want to (that's a separate decision though).